### PR TITLE
⚡ Bolt: [performance improvement] fast-path log parsing

### PR DESCRIPTION
**Pull request by rfv-code**
💡 What: The optimization implements a fast-path string extraction (`/^{"ts":"([^"]+)"/`) to parse log timestamp without doing `JSON.parse` on the whole line if the timestamp falls before `--since`.
🎯 Why: Reading a massive `events.jsonl` file and evaluating `JSON.parse()` on thousands of lines just to skip them later introduces unnecessary GC and CPU overhead.
📊 Impact: Considerably faster `getStats` or log reads when limiting logs with `--since`.
🔬 Measurement: Run the test suite (`vitest run skills/doc-wiki/scripts/tests/event_logger.test.ts`) and optionally profile the `event_logger`'s read performance when filtering out millions of older records.

---
*PR created automatically by Jules for task [13313259636941311736](https://jules.google.com/task/13313259636941311736) started by @rfv-code*

### CODE REVIEW

**Review by rfv-code — COMMENTED** [review]
## Review — ⚡ Bolt: fast-path log parsing (#36)

**Verdict: 🟢 Approve with minor notes.** The optimization is correct and, importantly, fails *safe*: when the anchored regex `/^{"ts":"([^"]+)"/` does not match, `fastPathSuccess` stays `false` and the line falls through to the existing `JSON.parse` + strict `G-EVENTS-TS-STRICT` path. Behavior is preserved; only the old-line skip path is accelerated.

### Findings

🟢 **Good** — `skills/doc-wiki/scripts/event_logger.ts:197` — Anchored `^{"ts":"` is the right call vs. the prior `.includes()` approach (which the `.jules/bolt.md` entry correctly calls out as fragile). `JSON.stringify` emits no inter-token whitespace, so the anchor holds for all logger-written lines.

🟢 **Good** — NaN handling in the fast path (`Number.isNaN(entryMs) || entryMs < sinceMs → continue`) matches the slow-path `G-EVENTS-TS-STRICT` semantics, so the drop-on-unparseable behavior is consistent across both paths.

🟡 **Minor** — `event_logger.ts:218` — unrelated reformat: the `entryMs` ternary was collapsed onto one line. Harmless but outside the change's scope.

🟡 **Minor** — No test was added. Existing `event_logger.test.ts` `--since` cases cover behavioral equivalence, but a line where `ts` is *not* the first key (forcing the fallback path) is the one case worth pinning explicitly. Optional.

🟠 **Process** — This overlaps PR #35 ("Fast-path timestamp check in event_logger"), also touching `_readEvents` `--since` filtering in the same files. #36 appears to supersede #35's `.includes()` approach with the anchored regex. Recommend the maintainer merge **one** and close the other to avoid a conflicting double-merge.

| # | Severity | Item | Action |
|---|----------|------|--------|
| 1 | 🟡 | Stray reformat at :218 | optional revert |
| 2 | 🟡 | No explicit fallback-path test | optional |
| 3 | 🟠 | Duplicate of #35 | pick one, close other |

File: skills/doc-wiki/scripts/event_logger.ts [hunk 1]
```diff
@@ -194,11 +194,35 @@ function _readEvents(
   for (const rawLine of raw.split("\n")) {
     const line = rawLine.trim();
     if (!line) continue;
+
+    if (sinceMs !== null && !Number.isNaN(sinceMs)) {
+      // ⚡ Bolt: Fast-path string check to avoid JSON.parse on old log lines.
+      // Use strict, anchored regex check to pre-filter relevant lines before
+      // parsing, preventing fragile regex matches on inner JSON fields.
+      const match = line.match(/^{"ts":"([^"]+)"/);
```
[the remark this answers is not in the document]
Comment (rfv-code) [reply skills/doc-wiki/scripts/event_logger.ts:202]: Verified against current branch tip + repo conventions — false positive for doc-wiki. `events.jsonl` is only ever written by `event_logger.ts:158` via `fs.appendFileSync(p, JSON.stringify(entry) + "\n")` — compact JSON, no space after colon, `ts` first key (`event_logger.ts:149`). Per CLAUDE.md ("No Python in this repo") there is no `json.dumps` producer, so no spaced-prefix legacy lines exist. Non-conforming lines still parse correctly via the unconditional `JSON.parse` + G-EVENTS-TS-STRICT fallback. Resolved.

**Comment by rfv-code** [comment]
Closing as duplicate of #40. PR #40 has the broadest scope (timestamp regex fast-path + the matching `.jules/bolt.md` note) and is the most recent in the cluster of daily Bolt re-proposals on `event_logger` JSON-parsing optimization. If this PR carries an edge case or test #40 doesn't (e.g. a specific regression case), please cherry-pick onto the #40 branch.
